src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.TryGetProperty.cs

```diff
@@ -176,7 +176,7 @@ internal bool TryGetNamedPropertyValue(int index, ReadOnlySpan<byte> propertyNam
 
                         // If everything up to where the property name has a backslash matches, keep going.
                         if (propertyName.Length > idx &&
-                            currentPropertyName.Slice(0, idx).SequenceEqual(propertyName.Slice(0, idx)))
+                            currentPropertyName.StartsWith(propertyName.Slice(0, idx)))
```

Comment:
I think the original code was probably a bit easier to understand. The new code is a bit shorter. Unclear which one is better.

Reply:
eye of the beholder I guess. since there's disagreement on which is clearer, I'll just close it.